### PR TITLE
Removes Princess of Terror RNG

### PR DESCRIPTION
**Pull request by PopeDaveThe3th**
## What Does This PR Do
Princesses of Terror now gain a guaranteed egg every 2 minutes , replacing the old system of a 25% chance for an egg every 30 seconds.

## Why It's Good For The Game
RNG is foul and wretched. Consistency is good.

## Testing
- spawned as TPrincess
- Waited for two minutes
- egg

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  <hr>
Screenshot 2024-09-26 122038

Screenshot 2024-09-26 122050

## Changelog

:cl:
tweak: Princess eggs are no longer RNG.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 1]
```diff
@@ -34,8 +34,8 @@
 
 	canlay = 0
 	hasnested = TRUE
-	spider_spawnfrequency = 300 // 30 seconds
-	var/grant_prob = 25 // 25% chance every spider_spawnfrequency seconds to gain 1 egg
+	spider_spawnfrequency = 120 SECONDS
+	var/grant_prob = 100 // eggs are guaranteed. I HATE RNG.
```
Comment (1080pCat) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:38]: could you just remove `grant_prob` if you're setting it to 100?

Comment (PopeDaveThe3th) [reply code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:38]: Probably doable but I'm a terrible coder and doing so breaks other things.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 2]
```diff
@@ -100,8 +96,7 @@
 		to_chat(src, "<span class='notice'>You have [canlay] eggs available to lay.</span>")
 
 /mob/living/simple_animal/hostile/poison/terror_spider/queen/princess/show_egg_timer()
-	var/average_timer = (1 / (grant_prob / 100)) * (spider_spawnfrequency / 10)
-	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [average_timer] seconds, on average.</span>")
+	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every 120 seconds, on average.</span>")
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:99]:
```suggestion
	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [spider_spawnfrequency /10] seconds, on average.</span>")
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
One last thing

**Review by ErdGinalD — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 3]
```diff
@@ -100,8 +96,7 @@
 		to_chat(src, "<span class='notice'>You have [canlay] eggs available to lay.</span>")
 
 /mob/living/simple_animal/hostile/poison/terror_spider/queen/princess/show_egg_timer()
-	var/average_timer = (1 / (grant_prob / 100)) * (spider_spawnfrequency / 10)
-	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [average_timer] seconds, on average.</span>")
+	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [spider_spawnfrequency /10] seconds, on average.</span>")
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:99]:
```suggestion
	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [spider_spawnfrequency / 10] seconds, on average.</span>")
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm [hunk 4]
```diff
@@ -100,8 +96,7 @@
 		to_chat(src, "<span class='notice'>You have [canlay] eggs available to lay.</span>")
 
 /mob/living/simple_animal/hostile/poison/terror_spider/queen/princess/show_egg_timer()
-	var/average_timer = (1 / (grant_prob / 100)) * (spider_spawnfrequency / 10)
-	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [average_timer] seconds, on average.</span>")
+	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [spider_spawnfrequency / 10] seconds, on average.</span>")
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/princess.dm:99]: It's not an average anymore.
```suggestion
	to_chat(src, "<span class='danger'>Too soon to attempt that again. You generate a new egg every [spider_spawnfrequency / 10] seconds.</span>")
```

**Review by Contrabang — APPROVED** [review]

**Pull request merged**